source/ed/data.cpp

```diff
@@ -600,45 +600,48 @@ void Data::pick_up_drop_of_on_borders() {
-         *                           |- Stay-In -|
+         *      Example 1:
+         *      ----------
+         *                                            (1)  (2)
+         *              out          in   out         in   out         in   out         in
+         *               X    SP1    |    ▲    SP2    |    ▲    SP3    |    ▲   SP4     X
+         *               X           ▼    |           ▼    |           |    |           X
+         *  (prev) VJ:1   08:00-09:00      10:00-11:00     |           ▼    |           X
+         *  (next) VJ:2                                     12:00-13:00      14:00-15:00
+         *                                |---------- Stay In ---------|
+         *
+         *       Example 1 is the only case were we allow specific pick-up and drop-off
          *
-         *   Example 2:
-         *   ----------
-         *                                       (1)  (2)
-         *         out          in   out         in   out         in   out         in
-         *          X    SP1    |    ▲    SP2    |    ▲    SP3    |    ▲   SP4     X
-         *          X           ▼    |           ▼    |           |    |           X
-         *    VJ:1   08:00-09:00      10:00-11:00     |           ▼    |           X
-         *    VJ:2                                     12:00-13:00      14:00-15:00
-         *                           |---------- Stay In ---------|
          *
-         *  Example 2 is the only case were we allow specific pick-up and drop-off
+         *      Example 2:
+         *      ----------
+         *              out          in   out         in
+         *               X    SP1    |    ▲    SP2    X
+         *               X           ▼    |           X
+         *  (prev) VJ:1   08:00-09:00      10:00-11:00
+         *  (next) VJ:2                    10:00-11:00      14:00-15:00
+         *                                X           ▲    |           X
+         *                                X           |    ▼   SP3     X
+         *                                out         in   out         in
+         *                                |- Stay-In -|
          */
-        if (vj->prev_vj) {
-            if (vj->joins_on_different_stop_points(*vj->prev_vj)) {
-                vj->prev_vj->stop_time_list.back()->pick_up_allowed = true;  // (1) in example 2
-                first_st->drop_off_allowed = true;                           // (2) in example 2
-            }
+
+        if (vj->next_vj and vj->joins_on_different_stop_points(*vj->next_vj)) {
+            // example 1 VJ:1
+            first_st->drop_off_allowed = false;
+        } else if (vj->prev_vj and vj->joins_on_different_stop_points(*vj->prev_vj)) {
+            // example 1 VJ:2
+            last_st->pick_up_allowed = false;
+        } else {  // no stay-in or example 2
+            first_st->drop_off_allowed = false;
+            last_st->pick_up_allowed = false;
```

Comment:
It seems to me you have nothing to do in case of example 1 (join on different stop_point) and when there is no stay-in.
The only case where we want to forbid something is when the join is on the same stop_point, right ?

```suggestion
       // In example 2 (join on same stop_point), we disable the pickup at the last stop of VJ:1 (prev_vj)
       //  and we disable the drop_off at the first_stop of VJ:2 (vj)
        if (vj->prev_vj and ! (vj->joins_on_different_stop_points(*vj->prev_vj)) ) {
            prev_vj->stop_time_list.back()->pick_up_allowed = false;
            vj->stop_time_list.front()->drop_off_allowed=false;

```

Reply:
For example 1 (a stay-in on different stops), you still need to forbid the extremity of the journey : 
 * forbid drop-off on the first stop of the left-hand VJ
 * forbid pickup on the last stop of the right-hand VJ

The problem occurs at the junction of the stay-in ( for `in (1)` and  `out (2)` example 1), where we were previously forcing pickup and drop-off to `true`. We now leave the data to define it (which is `true` by default`)